substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -0,0 +1,263 @@
+    pub struct Entity<AccountId> {
+        pub version: u32,
+        pub id: u32,
+        pub name: Vec<u8>,
+        pub account_id: AccountId,
+        pub country: Vec<u8>,
+        pub city: Vec<u8>,
+    }
+
+    #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, Debug, TypeInfo)]
+    pub struct Node<PubConfig, If> {
+        pub version: u32,
+        pub id: u32,
+        pub farm_id: u32,
+        pub twin_id: u32,
+        pub resources: Resources,
+        pub location: Location,
+        pub country: Vec<u8>,
+        pub city: Vec<u8>,
+        // optional public config
+        pub public_config: Option<PubConfig>,
+        pub created: u64,
+        pub farming_policy_id: u32,
+        pub interfaces: Vec<If>,
+        pub certification: NodeCertification,
+        pub secure_boot: bool,
+        pub virtualized: bool,
+        pub serial_number: Vec<u8>,
+        pub connection_price: u32,
+    }
+
+    #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, Debug, TypeInfo)]
+    pub struct Location {
+        pub longitude: Vec<u8>,
+        pub latitude: Vec<u8>,
+    }
+}
+
+pub mod v10 {
+    use super::*;
+    use crate::Config;
+
+    use frame_support::{pallet_prelude::Weight, traits::OnRuntimeUpgrade};
+    use sp_std::marker::PhantomData;
+    pub struct GridMigration<T: Config>(PhantomData<T>);
+
+    impl<T: Config> OnRuntimeUpgrade for GridMigration<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<(), &'static str> {
+            assert!(PalletVersion::<T>::get() == types::StorageVersion::V9Struct);
+
+            info!("👥  TFGrid pallet to v10 passes PRE migrate checks ✅",);
+            Ok(())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            migrate::<T>()
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade() -> Result<(), &'static str> {
+            assert!(PalletVersion::<T>::get() == types::StorageVersion::V10Struct);
+
+            info!(
+                "👥  TFGrid pallet migration to {:?} passes POST migrate checks ✅",
+                Pallet::<T>::pallet_version()
+            );
+
+            Ok(())
+        }
+    }
+}
+
+fn migrate<T: Config>() -> frame_support::weights::Weight {
+    if PalletVersion::<T>::get() == types::StorageVersion::V9Struct {
+        migrate_entities::<T>() + migrate_nodes::<T>() + update_pallet_storage_version::<T>()
+    } else {
+        info!(" >>> Unused migration");
+        0
+    }
+}
+
+fn migrate_entities<T: Config>() -> frame_support::weights::Weight {
+    info!(" >>> Migrating entities storage...");
+
+    let mut migrated_count = 0;
+
+    // We transform the storage values from the old into the new format.
+    Entities::<T>::translate::<deprecated::Entity<AccountIdOf<T>>, _>(|k, entity| {
+        info!("     Migrated entity for {:?}...", k);
+
+        let country_name_input: CountryNameInput = BoundedVec::try_from(entity.country).unwrap();
```

Comment:
This feels weird to me, we should not unwrap here. If a panic is thrown here, the migration will be blocked and the chain will halt.

Reply:
I replaced the `unwrap()` by more elegant solution to avoid panic during migration:
- using functions that returns an error as suggested on your below [comment](https://github.com/threefoldtech/tfchain/pull/441#discussion_r995413652)
- defining `Default` trait for fields
- using `unwrap_or_default()` instead of `unwrap()` when no other choice